hermeto/core/package_managers/pip/rust.py

```diff
@@ -24,70 +23,78 @@ def _has_rust_build_deps(raw_build_dependencies: list[str]) -> bool:
         # File name and package name may differ e.g. when there is a hyphen in
         # package name it might be replaced by an underscore in a file name.
-        pname = Path(Path(fname).name)
-        while pname.suffix in (".tar", ".gz", ".tgz"):
-            pname = pname.with_suffix("")
-        tf = tarfile.open(fname)
-        toplevel_cargo = f"{pname}/Cargo.toml"
-        try:
-            tf.getmember(toplevel_cargo)
-            rust_root = pname
-        except KeyError:
-            # only skip if no Cargo.toml is present in the package
-            if not any([fname.endswith("Cargo.toml") for fname in tf.getnames()]):
-                continue
-            # considering it has a Cargo.toml, let's check if it depends on the typical toolchain
-            # for python+rust to rule out false positives
-            if not _depends_on_rust(tf):
-                continue
-            # find the top-most Cargo.toml in the package - that's not necessarily the most accurate
-            # solution, but this heuristic has proven to work on the most popular python packages
-            # that have rust dependencies; if this stops working, then we would probably need to check
-            # pyproject toml config section for maturin or setuptools-rust...
-            # More info on this issue in the design doc
-            # https://github.com/hermetoproject/hermeto/blob/e5fa5c0fcd0dff62cf02be5b0d219e04c1ea440c/docs/design/cargo-support.md#L806
-            cargo_manifests = [name for name in tf.getnames() if name.endswith("Cargo.toml")]
-            rust_root = Path(sorted(cargo_manifests, key=len)[0]).parent
-
-        tf.extractall(path=Path(str(tf.name)).parent, filter="data")
-        packages_containing_rust_code.append(CargoPackageInput(type="cargo", path=rust_root))
+        package_path = p.get("path", "")
+        if not package_path:
+            continue
+
+        filename = Path(Path(package_path).name)
+        extract_filter = None if filename.suffix == ".zip" else "data"
+        while filename.suffix in (".zip", ".tar", ".gz", ".tgz"):
+            filename = filename.with_suffix("")
+
+        pip_deps_dir = Path(package_path).parent
+        extract_dir = pip_deps_dir / filename
+        shutil.unpack_archive(package_path, extract_dir=extract_dir, filter=extract_filter)  # type: ignore[arg-type]
+
+        cargo_files = list(extract_dir.rglob("Cargo.toml"))
+        if not cargo_files:
+            shutil.rmtree(extract_dir, ignore_errors=True)
+            continue
+
+        # The unpacked URL/VSC package may have an arbitrary directory name that we cannot control.
+        # Therefore, it is inside a predictable directory derived from the package name.
+        subdirs = [d for d in extract_dir.iterdir() if d.is_dir()]
+        if len(subdirs) == 1:
+            source_dir = subdirs[0]
+        else:
+            # should not happen
+            source_dir = extract_dir
+
+        if not _depends_on_rust(source_dir):
```

Comment:
A minor, optional, non-blocking naming suggestion: s/not _depends_on_rust/no_dependency_on_rust_detected_in/ (if accepted would require an inversion in the original logic).